integreat_cms/xliff/utils.py

```diff
@@ -319,39 +320,74 @@ def get_xliff_import_diff(request, xliff_dir):
                 page=page_translation.page,
                 language=page_translation.language,
             )
-            diff.append(
-                {
-                    "file": xliff_file,
-                    "existing": existing_translation,
-                    "import": page_translation,
-                    "source_diff": {
-                        "title": "\n".join(
-                            list(
-                                difflib.unified_diff(
-                                    [existing_translation.title],
-                                    [page_translation.title],
-                                    lineterm="",
-                                )
-                            )[2:]
+            translation_key = get_translation_key(existing_translation)
+            if translation_key in diff:
+                # Show global error to indicate duplicate translation
+                messages.error(
+                    request,
+                    format_html(
+                        __(
+                            _(
+                                "Page <b>{}</b> from file <b>{}</b> was also translated in file <b>{}</b>."
+                            ),
+                            _(
+                                "Please check which of the files contains the most recent version and upload only this file."
+                            ),
+                            _(
+                                "If you confirm this import, only the first file will be imported and the latter will be ignored."
```

Comment:
Can we guarantee ordering? I'm not sure.

Reply:
I think not. https://docs.python.org/3/library/glob.html#module-glob states in the first sentence:

> results are returned in arbitrary order

Reply:
I'd suggest that we simply call `sort()` when iterating over the returned list.

Reply:
Indeed, in my tests I also noticed that the order is arbitrary. I meant "first" in relation to the position in the sentence. So in the message "Page X from file Y was also translated in file Z.", I interpreted "Y" as the "first" file, and "Z" as the latter...
Do you have a better idea for the wording? Maybe "Page X was translated in two files, only file Y will be imported and file Z will be ignored"?

Reply:
Calling `sort()` won't change much, since if the user interprets "first" in relation of the file in the zip file, they will open their zip file and think that the first file which appears is the first file. But if we call sort, this is not the case.

Reply:
Okay, we could just rephrase to `One of the following files will not be imported` ?

Reply:
But I mean we already know that the existing diff entry `diff[translation_key]["file"]` will be imported and the file `xliff_file` which comes later in the loop will be ignored. I tried to phrase the sentence in a way that points that out. Do you have a better wording in mind?

Reply:
Ahh wait, I'm stupid, sorry 😁
I never experienced differences in the order between the diff calculation and the actual import, but yes, according to the docs the order might differ. So yes I agree that it makes sense to run sort on the list to make the order deterministic and we're able to tell the user which version will be imported...

Reply:
Alright, thanks for figuring it out :)